processing/src/main/java/io/druid/segment/filter/Filters.java

```diff
@@ -363,34 +386,47 @@ static double estimateSelectivityOfBitmapTree(
    * <p>
    * The non-overlap ratio can be computed like below.
    * <p>
-   * nonOverlapRatio(b1, b2) = size(union(b1, b2)) / (size(b1) + size(b2))
+   * overlapSize = size(b1) + size(b2) - size(union(b1, b2))
+   * nonOverlapRatio(b2) = (size(b2) - overlapSize) / size(b2)
+   * <p>
+   * The approximate unioned size is
+   * <p>
+   * unionedSize = size(b1) + size(b2) * nonOverlapRatio(b2)
    * <p>
    * Given bitmaps, this method calculates the non-overlap ratios of N bitmap samples,
    * and then returns the average of them.
    *
    * @param bitmapIndex   bitmap index to retrieve bitmaps
    * @param bitmapIndexes a list of indexes of bitmaps
    *
-   * @return approximated average non-overlap ratio of bitmaps
+   * @return approximated average non-overlap ratio of bitmaps.
    *
    * @see #estimatePredicateSelectivity(ColumnSelector, String, BitmapIndexSelector, Predicate)
    */
-  private static double computeNonOverlapRatioFromRandomBitmapSamples(
+  static double computeNonOverlapRatioFromRandomBitmapSamples(
```

Comment:
I vote for disabling this feature now not because the model yields very imprecise results (it's ok as long as there is no regression compared to the state prior to this PR), but because it generates a lot of garbage. The default bitmap algorithm is concise. Pairwise union of 100 bitmaps creates 100 distinct bitmaps, which immediately become garbage. Union of 100 bitmaps at once creates only one resulting bitmap.